### PR TITLE
LOOP-1872 Allow dose data without HealthKit write access

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-1872
- https://tidepool.atlassian.net/browse/LOOP-2184
- Initialize DoseStore with provenance identifier
- Authorize HealthKit and initiate observer query on every run

### CODE REVIEW

**Comment by darinkrauss** [comment]
LW PR: https://github.com/tidepool-org/LoopWorkspace/pull/322

File: Loop Status Extension/StatusViewController.swift [hunk 1]
```diff
@@ -92,7 +92,8 @@ class StatusViewController: UIViewController, NCWidgetProviding {
         observationEnabled: false,
         insulinModel: defaults?.insulinModelSettings?.model,
         basalProfile: defaults?.basalRateSchedule,
-        insulinSensitivitySchedule: defaults?.insulinSensitivitySchedule
+        insulinSensitivitySchedule: defaults?.insulinSensitivitySchedule,
+        provenanceIdentifier: HKSource.default().bundleIdentifier
```
Comment (ps2) [inline Loop Status Extension/StatusViewController.swift:96]: Does this end up with different provenanceIdentifiers for Loop and the extension? Should we be using something like the app group identifier?

**Review by ps2 — COMMENTED** [review]
Wondering about the bundle identifiers used here, and whether we should use the App group id instead.

File: Loop Status Extension/StatusViewController.swift (hunk 1 above)
Comment (darinkrauss) [reply Loop Status Extension/StatusViewController.swift:96]: This mirrors the behavior of what HealthKit would do if you added entries from the app, extension, watch extension, etc.  In our case, we don't actually write any data *real* data in any place other than the main app. (Yes, we do write "shadow" data in the watch extension, for example, in order to temporarily persist data for display only. For example, when the user enters carbs on the watch it goes through the main app, is written to the main app cache and HealthKit, and then, later, that data sent back to the watch and written to the watch cache as a "shadow copy" of the data in the main app. It is used for display purposes only.)

Comment (rickpasetto) [reply Loop Status Extension/StatusViewController.swift:96]: (good question, I'm curious too...)

**Review by rickpasetto — APPROVED** [review]
LGTM

**Review by ps2 — APPROVED** [review]
LGTM!

**Comment by ps2** [comment]
Hmm, the Loop Extension (today widget) does construct a DoseStore and does use `insulinOnBoard`. Will it get the right doses that are stored via Loop with these changes?

**Comment by darinkrauss** [comment]
@ps2 Yes, it works fine. The Loop Extension uses the same sqlite file as the main app in the shared app group directory. Also, I tested and it works as expected.

**Pull request merged**